quinn-proto/src/endpoint.rs

```diff
@@ -642,6 +639,16 @@ impl Endpoint {
         conn
     }
 
+    fn check_connection_limit(&mut self) -> Result<(), TransportError> {
```

Comment:
I've been considering that as well. I think the simplified API it produces is very appealing. I've hesitated because "limit to N connections" is probably a very reasonable common path to promote, but setting the config is probably not much easier than just rolling your own, as of this change. If removing `concurrent_connections` sounds good to you, then I'm on board.

Reply:
I think that sounds like a good idea. It would allow users to implement it easily, and it's probably fine to allow arbitrarily many simultaneous connections for users who don't bother limiting it.

Reply:
Okay, let's do it. @gretchenfrage are you able to pick this up? I'd suggest just replacing the first commit by just ripping out everything that has to do with connection limit, and then somewhere at the end tacking on a commit that allows the caller to get the `Endpoint`'s current `open_connections()`.

Reply:
Yep I'll do that.

Reply:
See comment below.